includes/class-coblocks-labs.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * Register CoBlocks Labs
+ *
+ * @package CoBlocks
+ */
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Main @@pkg.title Class
+ *
+ * @since 2.0.0
+ */
+class CoBlocks_Labs {
+	/**
+	 * This plugin's instance.
+	 *
+	 * @var CoBlocks_Labs
+	 */
+	private static $instance;
+
+	/**
+	 * Registers the plugin.
+	 *
+	 * @return CoBlocks_Labs
+	 */
+	public static function register() {
+		if ( null === self::$instance ) {
+			self::$instance = new CoBlocks_Labs();
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * The Constructor.
+	 */
+	public function __construct() {
+
+		add_action( 'init', array( $this, 'register_settings' ) );
+		add_action( 'init', array( $this, 'propagate_settings' ) );
+		add_action( 'enqueue_block_editor_assets', array( $this, 'editor_assets' ) );
+
+	}
+
+	/**
+	 * Go theme is active.
+	 *
+	 * @since 2.24.4
+	 * @return boolean
+	 */
+	public static function is_go_theme_active() {
+		$template_part      = explode( 'themes/', esc_attr( wp_get_theme( get_template() )->get_template() ) );
+		$active_template    = isset( $template_part ) && isset( $template_part[0] ) ? $template_part[0] : false;
+		$active_theme_is_go = 'go' === $active_template;
+		return $active_theme_is_go;
+	}
+
+	/**
+	 * Go theme is installed.
+	 *
+	 * @since 2.22.4
+	 * @return string
+	 */
+	public static function is_go_theme_installed() {
+		$go_theme_path   = get_theme_root() . '/go/index.php';
+		$is_go_installed = file_exists( $go_theme_path );
+		return $is_go_installed;
+	}
```

Comment:
I think these two functions can be replaced with a single function. We do something similar in nextgen.

```php
public static function is_go_active() {
	return defined( 'GO_VERSION' ) && 'go' === get_option( 'stylesheet' );
}
```

This should check that Go is both installed and active, although I think these is even a duplicate check. Something like this should be enough to ensure Go is installed and active.

```php
public static function is_go_active() {
	return defined( 'GO_VERSION' );
}
```